ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,192 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import '../../core/nuxeo-page-provider.js';
+import '../search/nuxeo-search-form-layout.js';
+import '../search/nuxeo-search-results-layout.js';
+import '../widgets/nuxeo-dialog.js';
+import '@polymer/paper-button/paper-button.js';
+
+{
+  /**
+   * `nuxeo-document-picker`
+   * @group Nuxeo UI
+   * @element nuxeo-document-picker
+   */
+  class DocumentPicker extends Nuxeo.Element {
+    static get template() {
+      return html`
+        <style include="nuxeo-styles">
+          nuxeo-dialog {
+            display: flex;
+            flex-direction: column;
+            justify-content: space-between;
+            height: 100%;
+            width: 100%;
+            /* TODO check if we can use something a bit bigger here */
+            max-height: 84%;
+            max-width: 768px;
+            padding: 8px;
+            overflow: auto;
+          }
+          #topContainer {
+            display: flex;
+            justify-content: flex-end;
+          }
+          iron-icon {
+            opacity: 0.6;
+          }
+          iron-icon:hover {
+            cursor: pointer;
+            opacity: 1;
+          }
+          #mainContainer,
+          #topContainer,
+          nuxeo-results-view {
+            margin: 0;
+            padding: 0;
+          }
+          nuxeo-results-view {
+            --nuxeo-results-view-height: calc(100vh - 520px - var(--nuxeo-app-top));
+          }
+        </style>
+        <nuxeo-dialog id="dialog" modal>
+          <div id="mainContainer">
+            <div id="topContainer">
+              <iron-icon icon="nuxeo:cross" on-tap="_close"></iron-icon>
+            </div>
+            <nuxeo-results-view
+              id="resultsView"
+              provider="[[provider]]"
+              page-size="[[pageSize]]"
+              params="[[params]]"
+              quick-filters="{{quickFilters}}"
+              schemas="[[schemas]]"
+              enrichers="[[enrichers]]"
+              search-name="[[searchName]]"
+              aggregations="{{aggregations}}"
+              visible
+              show-filters
+              opened
+              search-form="[[searchForm]]"
+              on-navigate="_onNavigate"
+              on-results-changed="_onResultsChanged"
+            ></nuxeo-results-view>
+          </div>
+          <div class="buttons">
+            <!-- TODO check if we can have these stuck to the bottom (absolute positioning? check creation wizard) -->
+            <paper-button noink class="secondary" dialog-dismiss>Cancel</paper-button>
+            <paper-button noink class="primary" on-tap="_onInsert" id="insertButton">Insert</paper-button>
```

Comment:
Now that I think about this, "Insert" as a label seems too specific to the use case where we want to "insert" an image in an HTML document. What if we want to make use of it in another context in the future? Shouldn't this have a more generic label, like "Select" for example?

Reply:
I agree.
As long as @bchauvin agrees that "Select" is clear enough. For context, this is the label  of the button that confirms the selection of the documents and, in this use case in particular, inserts the images in the editor.
(leaving also a note here to remember myself to update all the other "insert" references both in the picker and in the HTML editor element)

Reply:
we don't use i18n here ?

Reply:
More generically.. all methods, default labels, etc.. should be about picking/selecting as, like @Gabez0r said, it's up to the user of the element to decide what to do with the picked/selected document ..